### ISSUE TITLE
🐛 [firebase_auth] linkWithCredential no longer works with EmailAuthCredential - anonymous account upgrade no longer working

### ISSUE THREAD

**Issue by nohe427** [firebase/flutterfire#11703]
On September 15, 2023 [email enumeration protection](https://cloud.google.com/identity-platform/docs/admin/email-enumeration-protection) was enabled by default for **all new Firebase projects  created after that date** to boost security. When this setting is enabled, it removes the ability to link an anonymous account with an email address. Regular federated identity provider linking still works, but the upgrade path for anonymous accounts is no longer able to be upgraded to an email credential.

If you do need to upgrade anonymous accounts to an email, the only workaround until we land an SDK fix is to [disable email enumeration protection](https://cloud.google.com/identity-platform/docs/admin/email-enumeration-protection#disable) on your Firebase project's underlying Cloud project. Disabling email enumeration protection will lower the security of your project.

**Comment by darshankawar** [comment]
This maybe is similar / related to https://github.com/firebase/flutterfire/issues/11661

**Comment by nohe427** [comment]
That looks to be the same issue

**Comment by tomasbaran** [comment]
When will it be officially fixed with a new release. Any ETA please?

**Comment by nohe427** [comment]
> @nohe427 When will it be officially fixed with a new release. Any ETA please?

There are quite a few moving parts on this one but we are projecting to have the Android and iOS SDKs patched soon. Once those patches land, we will coordinate with the Flutter SDK to adopt the new Android and iOS SDKs.

Additionally, its important to note that enumeration protection is a very important security feature and while disabling the protecting in the interim may help unblock you, you should definitely consider re-enabling once the patches land.

**Comment by MirceaX2Mobile** [comment]
`await auth.signInWithCredential(credential);`

The sign in it's not working anymore while linking an anonymous account using Phone Authentication. You get this

![Screenshot 2023-10-20 at 18 31 14](https://github.com/firebase/flutterfire/assets/33491440/4ac9a728-7fe5-4966-bba6-6cf2d5fac49f)

**Comment by nohe427** [comment]
@MirceaX2Mobile - Can you repost the minimum reproducible code for this?

**Comment by nohe427** [comment]
@MirceaX2Mobile - From my testing and re-reading the issue history, the issue you are experiencing is different and I am not able to reproduce the behavior that you are seeing.

My assumption is that your code looks something like this:

```dart
FirebaseAuth auth = FirebaseAuth.instance;

await auth.verifyPhoneNumber(
  phoneNumber: phoneNumber,
  codeSent: (String verificationId, int? resendToken) async {
    // Update the UI - wait for the user to enter the SMS code
    String smsCode = 'xxxx';

    // Create a PhoneAuthCredential with the code
    PhoneAuthCredential credential = PhoneAuthProvider.credential(verificationId: verificationId, smsCode: smsCode);

    // Sign the user in (or link) with the credential
    await auth.signInWithCredential(credential);
  },
);
```

Instead you do not want to call signInWithCredential, but just call link like this:

```dart
FirebaseAuth auth = FirebaseAuth.instance;

await auth.verifyPhoneNumber(
  phoneNumber: phoneNumber,
  codeSent: (String verificationId, int? resendToken) async {
    // Update the UI - wait for the user to enter the SMS code
    String smsCode = 'xxxx';

    // Create a PhoneAuthCredential with the code
    PhoneAuthCredential credential = PhoneAuthProvider.credential(verificationId: verificationId, smsCode: smsCode);

    final userCredential = await FirebaseAuth.instance.currentUser
      ?.linkWithCredential(credential);
  },
);
```

This should link your phone authentication method to the anonymous user and return your new userCredential for the accounts. If this does not work for you, please create a minimum reproducible sample and open a new issue in the flutterfire repository.

**Comment by ashu2902** [comment]
@nohe427 I'm also facing the same issue. So in my case, this is the exact description of the problem:

1. After the anonymous sign-in, when I try to link the credentials of the anonymous account to PhoneAuth credentials.
2.  If the PhoneAuth credential is an already existing account, it throws an exception "credential-already-in-use".
3. in the catch block, I delete the anonymous account and use the signInWithCredentials(), which I receive in the catch block. (since the error is thrown by 'linkWithCredential()', I receive the credentials along with the token).
4. At the third step it started throwing an error: [firebase_auth/invalid-credential] The supplied auth credential is malformed, has expired, or is not currently supported.

Also, this is happening only on my iOS devices and it is working perfectly on Android.

**Comment by MirceaX2Mobile** [comment]
@nohe427 @ashu2902 @darshankawar @russellwheatley This is the exact retro I'm facing too.

**Comment by ashu2902** [comment]
> @nohe427 @ashu2902 @darshankawar @russellwheatley This is the exact retro I'm facing too.

Have you made any progress with this?

**Comment by MirceaX2Mobile** [comment]
> > @nohe427 @ashu2902 @darshankawar @russellwheatley This is the exact retro I'm facing too.
> 
> Have you made any progress with this?

Nope. I'm using an old version of firebase_auth: 4.6.2

**Comment by nohe427** [comment]
@MirceaX2Mobile & @ashu2902 - Could you kindly open a new issue for this?

**Comment by ashu2902** [comment]
> > > @nohe427 @ashu2902 @darshankawar @russellwheatley This is the exact retro I'm facing too.
> > 
> > 
> > Have you made any progress with this?
> 
> Nope. I'm using an old version of firebase_auth: 4.6.2

Could you please share you flutter and other firebase dependencies versions too?

**Comment by ashu2902** [comment]
> @MirceaX2Mobile & @ashu2902 - Could you kindly open a new issue for this?

https://github.com/firebase/flutterfire/issues/11841 DONE

**Comment by Lyokone** [comment]
Hello, the new version released yesterday contains a fix for this issue on Android and iOS. Let me know if it's working on your side.

**Comment by MirceaX2Mobile** [comment]
@Lyokone I can confirm that this is now fixed. Regarding [https://github.com/firebase/flutterfire/issues/11841]

**Comment by Lyokone** [comment]
Thanks I'll close this

**Comment by tomasbaran** [comment]
It is not fixed on my end. 

When disabled email enumeration protection it was working—which was the workaround. Since it was reported it's fixed and suggested to enable it back—that's what I did. And it's not working again. Still the same error: operation-not-allowed
